db/changes.go

```diff
@@ -171,6 +172,160 @@ func (db *Database) AddDocInstanceToChangeEntry(entry *ChangeEntry, doc *Documen
 	}
 }
 
+func (db *Database) buildRevokedFeed(singleChannelCache SingleChannelCache, options ChangesOptions, triggeredBy uint64) <-chan *ChangeEntry {
+	feed := make(chan *ChangeEntry, 1)
+	sinceVal := options.Since.Seq
+
+	queryLimit := db.Options.CacheOptions.ChannelQueryLimit
+	requestLimit := options.Limit
+
+	paginationOptions := options
+	paginationOptions.Since.Seq = 0
+	paginationOptions.Since.LowSeq = 0
+
+	if options.Since.TriggeredBy > 0 {
+		// Need to check if there has been a change since the triggeredByVal. If there has we need to start again from
+		// 0. Otherwise we can continue where we left off
+		highSeq, err := db.user.FindMostRecentRevocationSeq(singleChannelCache.ChannelName())
+		if highSeq > triggeredBy && err == nil {
+			paginationOptions.Since.Seq = options.Since.TriggeredBy
+		}
+	}
+
+	go func() {
+		defer base.FatalPanicHandler()
+		defer close(feed)
+		var itemsSent int
+		var lastSeq uint64
+
```

Comment:
I don't think it would hurt to clone the comment from changesFeed about how the pagination is being computed.